src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -769,8 +772,13 @@ public void deleteResultsAndSubmissionsOfParticipation(Long participationId, boo
         resultsToBeDeleted.forEach(result -> resultService.deleteResult(result, false));
         // Delete all submissions for this participation
         submissions.forEach(submission -> {
-            if (submission instanceof ProgrammingSubmission) {
+            if (submission instanceof ProgrammingSubmission programmingSubmission) {
                 coverageReportRepository.deleteBySubmissionId(submission.getId());
+                // This clears the build log entries and deletes them from the corresponding table
+                programmingSubmission.setBuildLogEntries(Collections.emptyList());
+                programmingSubmission.setResults(Collections.emptyList());
+                submissionRepository.save(programmingSubmission);
+                buildLogEntryRepository.deleteByProgrammingSubmissionId(submission.getId());
                 buildLogStatisticsEntryRepository.deleteByProgrammingSubmissionId(submission.getId());
             }
             submissionRepository.deleteById(submission.getId());
```

Comment:
This block is repeated identically in the `SubmissionResource`. Would it make sense to move it to the `SubmissionRepository` and reuse it from there in both places?

Reply:
I see your point and have now moved it into the `ProgrammingSubmissionService`. The reason why I have not done so initially was because the code assumes that the results of the submission are empty - which is true in both usages but might not be true in general.
I now refactored it a bit and it should be fine now.